### PR TITLE
chore: Improve Test performance

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

Currently building can take upwards of 3 minutes. Running mock tests have no dependency on compilation. In order to reduce waiting this PR splits the two tests into their own files

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Test locally on your own infrastructure
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation)) 
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: .github/workflows/test_build.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    name: Resolve Modules
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v3
+      - id: set-matrix
+        run: ./scripts/resolve-modules.sh
+  test_unit:
```
Comment (erezrokah) [inline .github/workflows/test_build.yml:25]:
```suggestion
  test_build:
```

File: .github/workflows/test_build.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+name: Unit Tests
```
Comment (erezrokah) [inline .github/workflows/test_build.yml:1]:
```suggestion
name: Build
```

File: .github/workflows/test_build.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+      - name: Build
+        if: steps.changed-files.outputs.any_changed == 'true' || github.event_name != 'pull_request'
+        run: go build .
+        working-directory: ${{ matrix.workdir }}
+
+      - name: Test
+        if: steps.changed-files.outputs.any_changed == 'true' || github.event_name != 'pull_request'
+        run: go test -v ./...
+        working-directory: ${{ matrix.workdir }}
```
Comment (erezrokah) [inline .github/workflows/test_build.yml:93]:
```suggestion

      - name: Test
        if: steps.changed-files.outputs.any_changed == 'true' || github.event_name != 'pull_request'
        run: go test -v ./...
        working-directory: ${{ matrix.workdir }}
```

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @bbernays, I think the new workflow requires a bit of clean up.
We'll have to measure if it's worth spinning up more machines just for the build step as that has an overhead too

File: .github/workflows/test_build.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v3
+      - id: set-matrix
+        run: ./scripts/resolve-modules.sh
+  test_build:
+    name: Unit Tests
```
Comment (erezrokah) [inline .github/workflows/test_build.yml:26]:
```suggestion
    name: Build
```

**Comment by bbernays** [comment]
Building the AWS Plugin takes longer then the setup + mock testing so because of that fact the total time of this workflow can be reduced by 50%

**Review by erezrokah — APPROVED** [review]

**Pull request merged**